### PR TITLE
[paths] Moving limits and conditions to accurately reflect full paths

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- address #6800 by considering number of full paths rather than edges with limit
- The problem currently is that the edge_limit is causing later steps in a path to be excluded from the final visualization. So a person who went from `step 1 -> step 2 -> step 3 -> step 4` might be missing the `step 3 -> step 4`. This edge isn't considered dangling, but since it's not shown, the visualization denotes this user as dropping off. However, our path person query does a true query without limits and will return null on the person modal since it doesn't designate this user as having dropped off
- My solution is to remove some of the advanced controls based on edge weights. Change the limiting to be based on the number of full paths. There's a bit of guesstimating in terms of number of edges but as long as there's a `step count + path limit` provided, assuming maximum variability in path types, there will be `step count * path limit` possible edges. Most cases should prove to be much less so I'm not too concerned about overwhelming number of edges on the visualization
*If this affects the frontend, include screenshots.*  

## How did you test this code?
*Please describe.*
- update tests

### CODE REVIEW

File: ee/clickhouse/queries/test/test_paths.py [hunk 1]
```diff
@@ -1866,7 +1867,7 @@ def test_path_removes_duplicates(self):
             ],
         )
 
-    @test_with_materialized_columns(["$current_url"])
+    @test_with_materialized_columns(["$current_url", "$screen_name"])
```
Comment (EDsCODE) [inline ee/clickhouse/queries/test/test_paths.py:1870]: Somehow this just started to fail when it should've been failing all along

Comment (macobo) [reply ee/clickhouse/queries/test/test_paths.py:1870]: Might have been a test ordering thing? Mat columns were not cleaned up between tests, fixed in a in-progress PR

File: ee/clickhouse/queries/paths/paths.py [hunk 2]
```diff
@@ -205,13 +214,17 @@ def get_session_threshold_clause(self) -> str:
...
 
         if self._filter.end_point and self._filter.start_point:
             params.update({"target_point": self._filter.end_point, "secondary_target_point": self._filter.start_point})
             return (
                 """
+            , indexOf(paths, %(secondary_target_point)s) as start_target_index
+            , indexOf(paths, %(target_point)s) as end_target_index
+            """,
+                """
             , indexOf(compact_path, %(secondary_target_point)s) as start_target_index
```
Comment (neilkakkar) [inline ee/clickhouse/queries/paths/paths.py:228]: I assume you intended to remove this form here, and use compact_path above?

Comment (neilkakkar) [reply ee/clickhouse/queries/paths/paths.py:228]: Nvm, looks alright

Comment (neilkakkar) [reply ee/clickhouse/queries/paths/paths.py:228]: Edit: This becomes problematic with wildcard start & end points? And path persons with the same

File: ee/clickhouse/sql/paths/path.py [hunk 3]
```diff
@@ -30,16 +30,17 @@
...
-                                groupArray(path_item) as paths
-                        FROM ({path_event_query})
-                        GROUP BY person_id
+                            SELECT * {target_index_clause} FROM (
+                                SELECT person_id,
+                                    groupArray(toUnixTimestamp64Milli(timestamp)) as timing,
+                                    groupArray(path_item) as paths
+                                FROM ({path_event_query})
+                                GROUP BY person_id
+                            ) {boundary_event_filter} {limit_clause}
```
Comment (neilkakkar) [inline ee/clickhouse/sql/paths/path.py:39]: I think `boundary_event_filter` should still be outside? (ref the point above^)

**Comment by neilkakkar** [comment]
I very much like the idea of making persons & paths in sync by applying the advanced options inside vs at the end.

However, why do we need to remove the advanced options to make this happen?

There is a troublesome trade-off we make here as well: To get the person<->paths counts in sync, our aggregate paths count go out of whack: Since we're choosing full paths of a few persons vs. full paths between edge A->B, which is I think more important on Paths. (You want to see **aggregate** counts between different paths, not paths of specific users. For the latter, you go for sess recordings).

I think a way to build data trust here is to explain that the Persons Modal is _all the people_, while the Paths counts are limited for visualisation (and you can use advanced options to see everything). cc: @clarkus for ideas on copy /  how best to represent this.
Thoughts?

**Comment by EDsCODE** [comment]
I need to think more on this. Maybe a solution to all of this is applying the limiting that we do on the main query somehow to the person query too. The advanced options do a simple limiting on the edges which becomes unaligned with the results that the person query is trying to surface. That's the main thing I'm concerned about because those numbers are out of sync.

**Comment by clarkus** [comment]
In the early paths visualization work, I explored the concept of collapsed events or groups of "other". The idea here is that there is more stuff than can be visualized in the current chart area. Hovering over the group would itemize its makeup. In some cases, these collapsed events might have subsequent path items (they're clipped). We could feasibly trigger some alternate state where we allow a "current path item" to be selected such that we can conditionally show extra context for that path item (subsequent steps). Thoughts?

Screen Shot 2021-11-17 at 11 43 13 AM

https://www.figma.com/file/PA3RZmpBykvBW1du69Kk0A/Prototypes?node-id=26%3A6118

**Pull request closed without merging**